### ISSUE TITLE
[BUG] scan throws KeyError: 'hits' when no results are found

### ISSUE THREAD

**Issue by samodelkinas** [opensearch-project/opensearch-py#313]
https://github.com/opensearch-project/opensearch-py/blob/05326ba5474c1a1f5c1975760a4e37037f967dda/opensearchpy/helpers/actions.py#L570 throws an exception when scroll is used and search results are empty (hits key is not within hits dictionary). 
For example when used with AWS wrangler opensearch.search method:
```
df = wr.opensearch.search(
        client=client,
        index='storage_v3',
        search_body={
            "query": {
                "term": {
                    "feed_id": {
                        "value": "20230307"
                    }
                    
                }
            }
        },
        is_scroll=True)
```
will cause exception:
  File "/opt/airflow/venv/wrangler/lib64/python3.8/site-packages/awswrangler/opensearch/_read.py", line 120, in search
    documents = [_hit_to_row(doc) for doc in documents_generator]
  File "/opt/airflow/venv/wrangler/lib64/python3.8/site-packages/awswrangler/opensearch/_read.py", line 120, in <listcomp>
    documents = [_hit_to_row(doc) for doc in documents_generator]
  File "/opt/airflow/venv/wrangler/lib64/python3.8/site-packages/opensearchpy/helpers/actions.py", line 570, in scan
    while scroll_id and resp["hits"]["hits"]:
KeyError: 'hits'
This can be handled, though would be nice to have empty results instead so that upstream library returns empty dataframe or something.

**Comment by ReinGrad** [comment]
That there is a problem with the opensearchpy library actions.py file. In particular, in line 570 there is a while loop that checks for scroll_id and non-empty search results in the hits key of the corresponding dictionary. However, when using the library with the AWS Wrangler opensearch.search method and the search query does not return results, the hits key is missing from the resp dictionary, which leads to a KeyError exception.

To reproduce the problem, one would have to use the AWS Wrangler opensearch.search method to execute a search query that returns no results, and also use the is_scroll=True argument to enable the scroll function. This would cause a while loop on line 570 actions.py and results in a KeyError exception.

A possible behavior in this case would be if the opensearchpy library handled the absence of the hits key in the resp dictionary and returned empty search results, instead of throwing a KeyError exception.

### Decisions

To fix the problem, opensearchpy libraries could change the while loop on line 570 to check for the hits key in the resp dictionary before trying to access it. If the hits key is missing, the loop may be completed and empty search results may be returned. This would prevent a KeyError exception from occurring and ensure that the library functions properly even if the search results are empty.

**Comment by wbeckler** [comment]
Go for it!

**Comment by saimedhi** [comment]
@samodelkinas @ReinGrad Please feel free to take up this issue and raise a PR with your proposed solution. Thanks :)

**Comment by dblock** [comment]
@samodelkinas any interest in writing a (failing) unit test for this?

**Comment by samodelkinas** [comment]
> @samodelkinas any interest in writing a (failing) unit test for this?

Given I'm only running a couple of years experience with Python, this may be a bit challenging, but, I'll give it a shot within several weeks.

**Comment by Djcarrillo6** [comment]
@dblock @saimedhi I'd like to request co-assignment on this issue. If it's okay with @samodelkinas of course, I believe I can open a PR for this issue that includes both a confirmatory failing unit test & a proposed solution. I can most likely have it ready for initial review in the next day or so.

**Comment by dblock** [comment]
Go for it @Djcarrillo6, no need to ask for permission!

**Comment by Djcarrillo6** [comment]
@dblock I wrote the code for the confirmatory failing unit test for this issue. Where would you suggest I place the unit test file within the project test folders/files?

**Comment by Djcarrillo6** [comment]
@dblock  If I'm not mistaken, I believe this issue can be closed based on my closed PR for it [#616 ](https://github.com/opensearch-project/opensearch-py/pull/616)
